runtime/stdlib.js

```diff
@@ -406,7 +406,18 @@ function caml_compare_val (a, b, total) {
       } else if (b instanceof MlBytes ||
                  (b instanceof Array && b[0] === (b[0]|0))) {
         return -1;
-      } else if (typeof a != "number") {
+      } else if ((typeof a === "number" || typeof a === "boolean" || typeof a === "string") &&
```

Comment:
This would permit `a` being a `number` and `b` being a `string`.  Is this something that we have to worry about?

Reply:
We need to consider every possible combination of types. I rewrote the function to fix the code / make the code clearer.